createAccountFragment/src/main/java/com/brandon/createaccountfragment/view/CreateAccountFragment.kt

```diff
@@ -0,0 +1,129 @@
+                    emailField,
+                    locationField
+                )
+                fields.forEach {
+                    it.clearFocus()
+                    it.isSelected = false
+                }
+                val textList = fields.map { it.text.toString() }
+                if (textList.all { it.isNotEmpty() }) {
+                    viewModel.createAccountClicked(
+                        textList[0],
+                        textList[1],
+                        textList[2],
+                        textList[3],
+                        textList[4]
+                    )
+                } else fieldsError?.isVisible = true
+            }
+        }
+    }
+
+    private fun updateState(uiState: CreateAccountViewModel.CreateAccountUiState) {
+        binding.apply {
+            when (uiState) {
+                is CreateAccountViewModel.CreateAccountUiState.Idle -> {
+                    progressSpinner.isVisible = false
+                    buttonText.isVisible = true
+                    firstNameField.setText(uiState.firstName)
+                    lastNameField.setText(uiState.lastName)
+                    emailField.setText(uiState.email)
+                    dateField.setText(uiState.dateOfBirth)
+                    locationField.setText(uiState.location)
+                    firstNameError.isVisible = uiState.isFirstNameError
+                    lastNameError.isVisible = uiState.isLastNameError
+                    emailError.isVisible = uiState.isEmailError
+                    dateError.isVisible = uiState.isDateError
+                    locationError.isVisible = uiState.isLocationError
+                }
+                is CreateAccountViewModel.CreateAccountUiState.Loading -> {
+                    progressSpinner.isVisible = true
+                    buttonText.isVisible = false
+                    firstNameField.setText(uiState.firstName)
+                    lastNameField.setText(uiState.lastName)
+                    emailField.setText(uiState.email)
+                    dateField.setText(uiState.dateOfBirth)
+                    locationField.setText(uiState.location)
+                    listOf(
+                        firstNameError,
+                        lastNameError,
+                        dateError,
+                        emailError,
+                        locationError,
+                        fieldsError
+                    ).forEach { it?.isVisible = false }
+                }
+                is CreateAccountViewModel.CreateAccountUiState.Loaded -> {
+                    NotificationDialogFragment(
+                        title = getString(R.string.core_success),
+                        body = getString(R.string.create_account_success_body),
+                        onDismiss = {
+                            // TODO FA-125 Add Navigation Between Fragments
+                        }).show(childFragmentManager, null)
+                }
+            }
+        }
+    }
```

Comment:
Set the binding to null onDestroy

Reply:
Did this, had to give it an elvis operator in the onViewCreated because the compiler wouldn't recognize that it was never null. it also wouldn't except a nullable view as the return type. I could also double bang here but figured that made less sense.

It also didn't care if I used an if block to see if it was null.